### PR TITLE
Variable OperationalVP Added to CommonObjectiveFactory.java

### PR DESCRIPTION
**Pull request by IllianiCBT**
### Current Implementation
`MekHQ/src/mekhq/campaign/mission/CommonObjectiveFactory.java` details the victory conditions (objectives) for the various built in AtB scenarios. All objectives have a fixed Operational Victory Point value of 1.

### Problem
This means that it is not possible to edit the value of objectives when compared to each other. For example, in one scenario you might have three objectives: 1 primary, 2 secondary. All objectives are valued identically, meaning completing both secondary objectives but losing the primary is still considered a victory.

### Solution
I added the integer `OperationalVP` to objective functions, where applicable. I updated all calls from the various objectives to include this variable. All instances default to 1, so nothing has been changed insofar as rewarding Operational VPs. This isn't an update to Operation VP rewards, but preparatory work for a future review.

### Examples of Use
The below is an example of use. `OperationalVP` for the below is actually set to 1 in this PR (as per current implementation), not 99.
<p align =center>
image
image
</p>

### A Note on StratCon
With the exception of the base attacks scenario (the one you get when the enemy is Invincible), StratCon doesn't appear to use AtB's scenarios and already has this functionality. The focus of this PR is to bring that functionality to AtB scenarios, as this allows a holistic review of Operational VPs, rather than restricting any review to StratCon only.

### CODE REVIEW

File: MekHQ/src/mekhq/campaign/mission/atb/scenario/AlliedTraitorsBuiltInScenario.java [hunk 1]
```diff
@@ -81,11 +81,11 @@ public void setObjectives(Campaign campaign, AtBContract contract) {
 
         String allyBotName = getContract(campaign).getAllyBotName();
 
-        ScenarioObjective destroyHostiles = CommonObjectiveFactory.getDestroyEnemies(contract, 100);
+        ScenarioObjective destroyHostiles = CommonObjectiveFactory.getDestroyEnemies(contract, 1,100);
```
Comment (Sleet01) [inline MekHQ/src/mekhq/campaign/mission/atb/scenario/AlliedTraitorsBuiltInScenario.java:84]: Needs a space.

File: MekHQ/src/mekhq/campaign/mission/atb/scenario/AceDuelBuiltInScenario.java [hunk 2]
```diff
@@ -141,8 +141,8 @@ public void setExtraScenarioForces(Campaign campaign, ArrayList<Entity> allyEnti
     public void setObjectives(Campaign campaign, AtBContract contract) {
         super.setObjectives(campaign, contract);
 
-        ScenarioObjective destroyHostiles = CommonObjectiveFactory.getDestroyEnemies(contract, 100);
-        ScenarioObjective keepFriendliesAlive = CommonObjectiveFactory.getKeepFriendliesAlive(campaign, contract, this, 100, false);
+        ScenarioObjective destroyHostiles = CommonObjectiveFactory.getDestroyEnemies(contract,1, 100);
```
Comment (Sleet01) [inline MekHQ/src/mekhq/campaign/mission/atb/scenario/AceDuelBuiltInScenario.java:144]: Needs a space.

File: MekHQ/src/mekhq/campaign/mission/atb/scenario/BaseAttackBuiltInScenario.java [hunk 3]
```diff
@@ -162,14 +162,14 @@ public void setExtraScenarioForces(Campaign campaign, ArrayList<Entity> allyEnti
...
+        ScenarioObjective destroyHostiles = CommonObjectiveFactory.getDestroyEnemies(contract, 1, 50);
         destroyHostiles.addForce(String.format("%s%s", contract.getEnemyBotName(), SECOND_ENEMY_FORCE_SUFFIX));
         ScenarioObjective keepFriendliesAlive = CommonObjectiveFactory.getKeepFriendliesAlive(campaign, contract, this,
-                50, false);
+                1, 50, false);
 
         ScenarioObjective preserveBaseUnits = null;
         if (!isAttacker()) {
-            preserveBaseUnits = CommonObjectiveFactory.getPreserveSpecificFriendlies(BASE_CIVILIAN_FORCE_ID, 3, true);
+            preserveBaseUnits = CommonObjectiveFactory.getPreserveSpecificFriendlies(BASE_CIVILIAN_FORCE_ID, 1,3, true);
```
Comment (Sleet01) [inline MekHQ/src/mekhq/campaign/mission/atb/scenario/BaseAttackBuiltInScenario.java:172]: Needs a space.

File: MekHQ/src/mekhq/campaign/mission/atb/scenario/ChaseBuiltInScenario.java [hunk 4]
```diff
@@ -140,9 +140,9 @@ public void setObjectives(Campaign campaign, AtBContract contract) {
         super.setObjectives(campaign, contract);
 
         ScenarioObjective destroyHostiles = isAttacker()
-                ? CommonObjectiveFactory.getBreakthrough(contract, this, campaign, 50,
+                ? CommonObjectiveFactory.getBreakthrough(contract, this, campaign, 1,50,
```
Comment (Sleet01) [inline MekHQ/src/mekhq/campaign/mission/atb/scenario/ChaseBuiltInScenario.java:143]: Needs a space.

**Review by Sleet01 — CHANGES_REQUESTED** [review]
Just needs a few formatting fixes, then good to go.

File: MekHQ/src/mekhq/campaign/mission/atb/scenario/ChaseBuiltInScenario.java (hunk 4 above)
Comment (IllianiCBT) [reply MekHQ/src/mekhq/campaign/mission/atb/scenario/ChaseBuiltInScenario.java:143]: Done

File: MekHQ/src/mekhq/campaign/mission/atb/scenario/BaseAttackBuiltInScenario.java (hunk 3 above)
Comment (IllianiCBT) [reply MekHQ/src/mekhq/campaign/mission/atb/scenario/BaseAttackBuiltInScenario.java:172]: Done

File: MekHQ/src/mekhq/campaign/mission/atb/scenario/AceDuelBuiltInScenario.java (hunk 2 above)
Comment (IllianiCBT) [reply MekHQ/src/mekhq/campaign/mission/atb/scenario/AceDuelBuiltInScenario.java:144]: Done

File: MekHQ/src/mekhq/campaign/mission/atb/scenario/AlliedTraitorsBuiltInScenario.java (hunk 1 above)
Comment (IllianiCBT) [reply MekHQ/src/mekhq/campaign/mission/atb/scenario/AlliedTraitorsBuiltInScenario.java:84]: Done

**Pull request merged**